bin/git-theta-filter

```diff
@@ -59,26 +68,43 @@ def clean(args):
     checkpoint_handler = checkpoints.get_checkpoint_handler()
     model_checkpoint = checkpoint_handler.from_file(sys.stdin.buffer)
 
+    update_serializer = params.get_update_serializer()
     prev_metadata = metadata.Metadata.from_commit(repo, args.file, "HEAD").flatten()
 
     async def _clean(param_keys, new_param):
         logging.debug(f"Cleaning {'/'.join(param_keys)}")
         param_metadata = prev_metadata.get(param_keys)
         new_tensor_metadata = metadata.TensorMetadata.from_tensor(new_param)
+
         # If the parameter tensor has not changed, just keep the metadata the same
```

Comment:
I think it is fine for now, but we might want to leave a todo about moving all this into the TensorMetadata class and then in the clean all we need to do is a `==` check.

Reply:
Yeah I struggled with how to do this. Ideally the `==` check would compute the hash distance and if needed load the actual tensors and double check with `np.allclose`. This means that this should probably go in `__eq__` for `ParamMetadata` since this has access to the LSH hash as well as the LFS metadata needed to actually materialize the tensor. However, materializing the tensor (the way it's currently implemented with update handlers) requires some extra information like the file name, param keys, etc.

This is possible but will probably require some refactoring that we should push to another issue.